### PR TITLE
secondlife/viewer-issues#72: Fix material preview making things disappear. General cleanup.

### PR DESCRIPTION
**Pull request by cosmic-linden**

### CODE REVIEW

File: doc/testplans/material_preview.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Material Preview
```
Comment (Geenz) [inline doc/testplans/material_preview.md:1]: I would probably ask someone like @Dan-Linden or @AtlasLinden to review the test plan to make sure it parses well.

**Review by Geenz — APPROVED** [review]
Generally looks good - though I think we need to revisit LLRenderTarget and the ref swapping at some point.  Not a blocker for merge though.

**Comment by cosmic-linden** [comment]
> Generally looks good - though I think we need to revisit LLRenderTarget and the ref swapping at some point. Not a blocker for merge though.

Agreed that the ref swapping is a bit odd. I wanted to avoid messing too much with the guts of the render pipeline for this more specialized thing for now.

**Comment by Geenz** [comment]
@cosmic-linden Fair.  I do want to get a pass at render targets specifically after featurettes - we'll figure that out after release though.

**Pull request merged**